cmd/pj-rehearse/main.go

```diff
@@ -111,7 +108,11 @@ func main() {
 
 	logger.Info("Rehearsing Prow jobs for a configuration PR")
 
-	prowConfig, err := prowconfig.Load(o.configPath, o.jobConfigPath)
+	if err := createMasterWorktree(o.candidatePath, jobSpec.Refs.BaseSHA); err != nil {
+		logger.WithError(err).Error("could not create master worktree")
+		gracefulExit(o.noFail)
+	}
+	prowConfig, err := prowconfig.Load(masterConfigPath, masterJobConfigPath)
```

Comment:
This won't work if `o.candidatePath` is not `.`

The `master` directory containing the base will be created within the `o.candidatePath` directory.
`createMasterWorktree` should IMO return the master worktree path (`string.Join(o.candidatePath, "master")`) and we should create `master{Job}ConfigPath` using the paths in the `diffs` package.

Reply:
`path.Join`